src/commonMain/kotlin/com/github/michaelbull/result/coroutines/SuspendableBinding.kt

```diff
@@ -16,11 +16,31 @@ suspend inline fun <V, E> binding(crossinline block: suspend ResultBinding<E>.()
         callsInPlace(block, InvocationKind.EXACTLY_ONCE)
     }
 
-    val receiver = ResultBindingImpl<E>()
+    val receiver = SuspendableResultBindingImpl<E>()
 
     return try {
         with(receiver) { Ok(block()) }
-    } catch (ex: BindException) {
-        receiver.error
+    } catch (ex: BindCancellationException) {
+        receiver.internalError
+    }
+}
+
+internal object BindCancellationException : CancellationException(null)
+
+@PublishedApi
+internal class SuspendableResultBindingImpl<E> : ResultBinding<E> {
+
+    lateinit var internalError: Err<E>
+
+    override fun <V> Result<V, E>.bind(): V {
+        return when (this) {
+            is Ok -> value
+            is Err -> {
+                if (::internalError.isInitialized.not()){
```

Comment:
ensures only the first failing bind to complete is reported as the error

Reply:
still not entirely sure if this the best way to guarantee this.
 given this change already requires coroutines core to be added, perhaps it would be better to use [`Mutex`](https://kotlinlang.org/docs/reference/coroutines/shared-mutable-state-and-concurrency.html#mutual-exclusion) for reading if its empty and writing to it.